pkgs/desktops/gnome-3/default.nix

```diff
@@ -399,6 +399,8 @@ lib.makeScope pkgs.newScope (self: with self; {
 
   gnome-packagekit = callPackage ./misc/gnome-packagekit { };
 
+  # TODO: remove this after 18.09 has forked off
+  gconf = throw "gconf has been removed because it’s a pile of trash, see https://github.com/NixOS/nixpkgs/pull/43268";
```

Comment:
Let's keep the error message for the end-user a bit... less opinionated :). I'm sure it **is** a piece of trash, but the message itself would be better suited as something like:

> gconf has been removed, it is deprecated. See https://github.com/NixOS/nixpkgs/pull/43268 for more information.

Couldn't find a solid date for the deprecation. Would have been nice for the message (it is deprecated since 20xx), showing how much of a pile of rot it is.

Additionally, since we're linking to the PR, it may be interesting to review the main body of #43268, not to remove the opinions, but to make it more obvious for the end-user what is the way forward.

Reply:
I was not able to find a date either, thought the wiki page for the initiative appears to have been created in 2009: https://wiki.gnome.org/action/recall/Initiatives/GnomeGoals/GSettingsMigration?action=recall&rev=1

Reply:
I added tl;dr to the OP in the PR.

Reply:
I'm :-1: on the "pile of trash" note. We may have users who _like_ gconf, or even _authored_ gconf. Being so judgemental gains us nothing risks hurting the feelings of our users, or being extra-annoying when they have to change their system configuration to keep up with NixOS.

What _would_ be useful here is a "what do I do now?" message about how to change their now-broken system in to a working-like-I-want system.

Reply:
Note, referring people to https://github.com/NixOS/nixpkgs/pull/43268 is not even especially helpful: it doesn't have any instructions about what the user should do, just that it is evidently feces.

Reply:
I did not really mean anything too bad by that, the fact is the majority of software is bad, including all of mine. Though I agree that the comment does not really add anything and some people might be offended by it so I will tone it down.